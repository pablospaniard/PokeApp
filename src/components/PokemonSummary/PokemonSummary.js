import React from 'react'
import PropTypes from 'prop-types'

import styles from './PokemonSummary.scss'
import {LinearProgress, Card, CardContent, Typography} from '@material-ui/core'

const PokemonSummary = props => {
  const {details, loading} = props
  const html = (
    <Card>
      <CardContent className={styles.Summary}>
        <Typography variant="subheading" style={{margin: 10}}>
          {` Height: ${details.height}, Weight: ${details.weight}`}
        </Typography>
        <Typography variant="subheading" style={{margin: 10}}>
          {` Base experience: ${details.base_experience}`}
        </Typography>
      </CardContent>
    </Card>
  )
  const content = !loading ? html : <LinearProgress color="secondary" />
  return content
}

PokemonSummary.propTypes = {
  details: PropTypes.object,
  loading: PropTypes.bool
}

export default PokemonSummary
