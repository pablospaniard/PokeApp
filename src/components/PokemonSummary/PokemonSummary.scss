@import '../../scss/vars.scss';
.Summary {
  background-color: $blue;
  &:hover {
    animation: summary 3s infinite;
    animation-direction: alternate;
    animation-timing-function: ease-in;
  }
}

@keyframes summary {
  0% {
    background-color: #4698cb;
  }
  100% {
    background-color: #ff4136;
  }
}
