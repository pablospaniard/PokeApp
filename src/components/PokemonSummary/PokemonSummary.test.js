import React from 'react'
import {shallow} from 'enzyme'
import PokemonSummary from './PokemonSummary'

describe('PokemonSummary will render properly', () => {
  const loading = false
  const details = {}

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<PokemonSummary details={details} loading={loading} />)
  })
  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot('PokemonSummary_snapshot_1')
  })
})
