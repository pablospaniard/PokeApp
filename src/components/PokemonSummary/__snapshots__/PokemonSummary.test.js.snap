// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonSummary_snapshot_1 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokemonSummary
    details={Object {}}
    loading={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(CardContent)
        className={undefined}
>
        <WithStyles(Typography)
                style={
                        Object {
                                "margin": 10,
                              }
                }
                variant="subheading"
        >
                 Height: undefined, Weight: undefined
        </WithStyles(Typography)>
        <WithStyles(Typography)
                style={
                        Object {
                                "margin": 10,
                              }
                }
                variant="subheading"
        >
                 Base experience: undefined
        </WithStyles(Typography)>
</WithStyles(CardContent)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            style={
                        Object {
                                    "margin": 10,
                                  }
            }
            variant="subheading"
>
             Height: undefined, Weight: undefined
</WithStyles(Typography)>,
          <WithStyles(Typography)
            style={
                        Object {
                                    "margin": 10,
                                  }
            }
            variant="subheading"
>
             Base experience: undefined
</WithStyles(Typography)>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " Height: undefined, Weight: undefined",
            "style": Object {
              "margin": 10,
            },
            "variant": "subheading",
          },
          "ref": null,
          "rendered": " Height: undefined, Weight: undefined",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " Base experience: undefined",
            "style": Object {
              "margin": 10,
            },
            "variant": "subheading",
          },
          "ref": null,
          "rendered": " Base experience: undefined",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(CardContent)
          className={undefined}
>
          <WithStyles(Typography)
                    style={
                              Object {
                                        "margin": 10,
                                      }
                    }
                    variant="subheading"
          >
                     Height: undefined, Weight: undefined
          </WithStyles(Typography)>
          <WithStyles(Typography)
                    style={
                              Object {
                                        "margin": 10,
                                      }
                    }
                    variant="subheading"
          >
                     Base experience: undefined
          </WithStyles(Typography)>
</WithStyles(CardContent)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              style={
                            Object {
                                          "margin": 10,
                                        }
              }
              variant="subheading"
>
               Height: undefined, Weight: undefined
</WithStyles(Typography)>,
            <WithStyles(Typography)
              style={
                            Object {
                                          "margin": 10,
                                        }
              }
              variant="subheading"
>
               Base experience: undefined
</WithStyles(Typography)>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " Height: undefined, Weight: undefined",
              "style": Object {
                "margin": 10,
              },
              "variant": "subheading",
            },
            "ref": null,
            "rendered": " Height: undefined, Weight: undefined",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " Base experience: undefined",
              "style": Object {
                "margin": 10,
              },
              "variant": "subheading",
            },
            "ref": null,
            "rendered": " Base experience: undefined",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
